cloudinit/config/cc_landscape.py

```diff
@@ -151,7 +151,7 @@ def handle(name: str, cfg: Config, cloud: Cloud, args: list) -> None:
             ]
         )
     )
-    subp.subp(["landscape-config", "--silent"] + cmd_params)
+    subp.subp(["landscape-config", "--silent", "--is-registered"] + cmd_params)
```

Comment:
I think we instead would want a pre-flight check of "landscape-config",  "--silent", "--is-registered" looking for a success exit 0. If we get exit zero, then emit a warning in logs that landscape-client is already registered. If exit 5, then we fall through to the full landscape-config --silent with the supplemental cmd_params.

Reply:
Agreed and implemented. What do you think? 

Unfortunately, integration tests are failing due to a new line character in "expected_calls" array. 
`call(['landscape-config', '--silent', '--is-registered'], rcs=[5]),\n call(['landscape-config', '--silent', '--computer-title', 'My" PC', '--data-path', '/var/lib/landscape/client', '--log-level', 'info', '--ping-url', 'http://landscape.canonical.com/ping', '--url', 'https://landscape.canonical.com/message-system'])`

I can't figure it out why the first "call" tuple of the array will be interpreted with a new line character. Any help is much appreciated. thx

Reply:
Nevermind, figured it out what was causing this issue and fixed it :)